### DISCUSSION TITLE
definition of usbh_pipe_t

### DISCUSSION THREAD

**Discussion by dkumaryadav26** [cherry-embedded/CherryUSB#201] [Q&A]
Hi All,
I am new to cherryUSB, please help me understand cherryUSB.

**Comment by sakumisu** [comment]
No such definition, please correct.

**Reply by dkumaryadav26** [reply]
Hi Sakumisu,
Thanks for the quick response.

I am getting a compilation error here in "int usbh_get_xhci_devaddr(usbh_pipe_t *pipe)"

**Reply by sakumisu** [reply]
XHCI port is not for you, please give up.
